components/Discussion/enhancers.js

```diff
@@ -526,7 +501,8 @@ mutation setDiscussionPreferences($discussionId: ID!, $discussionPreferences: Di
 
           // clone() the data object so that we can mutate it in-place.
           const data = JSON.parse(JSON.stringify(immutableData))
```

Comment:
probably a bit more efficient
```
const data = {
  ... immutableData,
  discussion: {
    ..immutableData.discussion,
    ..setDiscussionPreferences.userPreference
  }
}
```

Reply:
I do this cloning in a few other places in the enhancer file as well. In this case your proposed change would be possible. But in the other cases (fetchMore, new comment subscription) not because I don't statically know where to modify the whole Discussion object (could be at any depth of the comments tree). So I clone the whole Discussion object, find the place where to make the change, and modify the object in-place.